tests/conftest.py

```diff
@@ -329,6 +331,12 @@ def _get_contract(
 ):
     settings = Settings()
     settings.optimize = override_opt_level or optimize
+
+    if not version_check(begin="cancun"):
+        with pytest.raises(TransientStorageException):
+            compiler.compile_code(source_code)
+        return
```

Comment:
we could even do `pytest.skip` so we can avoid those annoying version checks after `get_contract()`

Reply:
the alternative might be to use something like `pytest.mark.transient`, and then we can modify the test collector to skip those for pre-cancun

Reply:
> the alternative might be to use something like pytest.mark.transient, and then we can modify the test collector to skip those for pre-cancun

does this mean we would not check for the same contracts that pre-cancun contracts using `transient` throws the correct exception?